visidata/loaders/archive.py

```diff
@@ -0,0 +1,73 @@
+import codecs
+import tarfile
+import zipfile
+
+from visidata import *
+
+
+class read_archive(Sheet):
+    'Provide wrapper around `zipfile` and `tarfile` libraries for opening archive files.'
+    rowtype = 'files'
+
+    def __init__(self, p, archive_type='zip'):
+        super().__init__(p.name, source=p)
+        self.archive_type = archive_type
+        if self.archive_type == 'zip':
+            self.columns = [
+                ColumnAttr('filename'),
+                ColumnAttr('file_size', type=int),
+                Column('date_time', type=date, getter=lambda col, row: datetime.datetime(*row.date_time)),
+                ColumnAttr('compress_size', type=int)
+            ]
+        if self.archive_type == 'tar':
+            self.columns = [
+                ColumnAttr('name'),
+                ColumnAttr('size', type=int),
+                ColumnAttr('mtime', type=date),
+                ColumnAttr('type', type=int),
+                ColumnAttr('mode', type=int),
+                ColumnAttr('uname'),
+                ColumnAttr('gname')
+            ]
+
+    def reload(self):
```

Comment:
also, now that I'm looking at this, we should try this with a very large tarfile.  I'm thinking we want the reload to be `@asyncthread` at minimum, and then start rows with an empty list and iterate over getmembers, calling addRow on each element.  The zip loader should really be doing the same thing, but it's less impactful because the .zip has a directory at the end of the file, and afaik .tar only has them spread throughout.

Reply:
Sound be taken care of in the last few commits.